From 72df36f945042b85386ea514a2affac71aec2e96 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=8D=C3=B1igo=20Huguet?= <ihuguet@redhat.com>
Date: Tue, 1 Apr 2025 08:43:24 +0200
Subject: [PATCH 1/1] core: ovs: fix NULL pointer dereference in ovsdb read
 timeout callback

Fixes: f7d321c6d6ab ('ovsdb: add watchdog for unparsable JSON data in socket')
(cherry picked from commit dc9bf255eeb4d21452227a590486deef78aac4f8)
(cherry picked from commit 9ec498f3210cbf51e2f02068046630aa524f152e)
---
 src/core/devices/ovs/nm-ovsdb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/devices/ovs/nm-ovsdb.c b/src/core/devices/ovs/nm-ovsdb.c
index e6d37d270c..2dbc842a73 100644
--- a/src/core/devices/ovs/nm-ovsdb.c
+++ b/src/core/devices/ovs/nm-ovsdb.c
@@ -2468,7 +2468,7 @@ again:
                  * content is broken (_json_read_msg() cannot extract any data) and
                  * we disconnect. */
                 priv->input_timeout_source =
-                    nm_g_timeout_add_seconds_source(5, _ovsdb_read_input_timeout_cb, NULL);
+                    nm_g_timeout_add_seconds_source(5, _ovsdb_read_input_timeout_cb, self);
             }
             return;
         }
-- 
2.49.0

